integration-testing/src/main/java/io/grpc/testing/integration/TestServiceImpl.java

```diff
@@ -293,7 +293,14 @@ private void scheduleNextChunk() {
         Chunk nextChunk = chunks.peek();
         if (nextChunk != null) {
           scheduled = true;
```

Comment:
```scheduled = true;``` should be moved inside the ```else {...}``` since that's where the job is actually getting scheduled using the executor. Immediate execution shouldn't result in scheduled being set to true, correct? As per the previous code, it was always scheduled hence setting true prior to scheduling the task.